protocol/x/ratelimit/ibc_middleware.go

```diff
@@ -0,0 +1,203 @@
+package ratelimit
+
+// TODO(CORE-854): Improve attribution message.
```

Comment:
Generally need to do attribution better since a lot of code is reused from Stride: https://en.wikipedia.org/wiki/Attribution_(copyright)

I have a call with legal team next week to align on what needs to be done here